ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java

```diff
@@ -458,7 +458,7 @@ PersistentQueryMetadata buildPersistentQueryInSharedRuntime(
         serviceContext
     );
 
-    return new PersistentQueriesInSharedRuntimesImpl(
+    return new BinPackedPersistentQueryMetadata(
```

Comment:
We should follow the naming convention for the QueryMetadata classes. I figured it was fine without the `Impl` since there's no interface to confuse it with, but I'm also fine with adding `Impl` if you have a preference. Not sure what the naming convention is when it comes to classses who don't strictly need the `Impl` suffix?

Reply:
`public class PersistentQueryMetadataImpl
    extends QueryMetadataImpl implements PersistentQueryMetadata` We do have this. I think its useful to have the `impl` because then when dealing out side the abstraction you can use `PersistentQueryMetadata` without having to check. But its not a hill i am willing to die on

Reply:
🤔  but in that example the `Impl` _is_ necessary due to the interface that would otherwise have the same name. Whereas here there's no other interface, the `BinPackedPersistentQueryMetadata` just directly implements `PersistentQueryMetadata`

But fair enough, I'll leave the `Impl` suffix in the name